src/components/Dropdown/Dropdown.tsx

```diff
@@ -786,9 +810,47 @@ const Dropdown = ({
     [rememberScrollPosition],
   );
 
-  const closeIcons = (
-    <>
-      {handleClear && values.length > 0 && (
+  const valueCountCloseIconHandler = () => {
+    if (showValueCount && showCloseIcon) {
```

Comment:
shouldn't these be split so showValueCount only shows/hides the value count? Otherwise why have two flags if both need to be true to see either of them